src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> item;
+    private Node<T> node;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        node = new Node<>();
+        node.value = value;
+        if (isEmpty()) {
+            head = node;
+        } else {
+            item = getNode(size - 1);
+            link(item, node, null);
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        item = getNode(index);
+        if (item == null) {
+            add(value);
+        } else {
+            node = new Node<>();
+            node.value = value;
+            link(item.prev, node, item);
+            if (head.equals(item)) {
+                head = node;
+            }
+            if (tail.equals(node)) {
+                tail = node;
+            }
```

Comment:
```suggestion
         if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        /*
        private void checkIndex(int index) {
            if (index >= size || index < 0) {
                 throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
             }
         } */
        Node<T> oldNode = getNode(index);
        Node<T> oldNodePrev = oldNode.prev;
        Node<T> newNode = new Node<>(oldNodePrev, value, oldNode);
        oldNode.prev = newNode;
        if (oldNodePrev == null) {
            head = newNode;
        } else {
            oldNodePrev.next = newNode;
        }
```
Якщо перевірити що індекс == size то можна добавити і далі по коду не іти 
перевірку на індекс було би зручно перенести в окремий метод 
ускладнювати if else неможна
Don’t complicate if-else construction.
[https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html](url)